telegram/client.py

```diff
@@ -382,6 +386,20 @@ def get_authorization_state(self) -> AsyncResult:
 
         return self._send_data(data, result_id='getAuthorizationState')
 
+    def _wait_authorization_result(self, result) -> str:
+        authorization_state = None
+        if result:
+            result.wait(raise_exc=True)
```

Comment:
I've added a new parameter to the `_send_data` function: `block`, it can be used in the `_send_password` and `_send_telegram_code` methods to prevent waiting here.

Reply:
roger that